templates/route_down-Debian.erb

```diff
@@ -10,3 +10,5 @@ if [ "$IFACE" = "<%= @interface -%>" ] || [ "$IFACE" = "--all" ]; then
   fi
 <%- end -%>
 fi
+# non-zero exit code causes ifupdown to leave the interface partly configured, which may result in connectivity loss
```

Comment:
Similarly to the above I'd prefer to add a parameter to the route define like $clean_exit_on_failure , undef by default. to allow users to decide the intended behaviour, then , possibly, in the define code set a variable like $real_clean_exit_on_failure than, in case the parameter is not set tries to do the sanest thing (ie set to true if ipv4 is not configured but ipv6 is) and then,  if this  $real_clean_exit_on_failure is true, then add the above 2 lines otherwise template remains as before